src/BaseApiAbstract.php

```diff
@@ -33,14 +33,18 @@
 
     const STRATEGY_AUTH = 'auth';
 
-    const STRATEGY_SEARCH = 'search';
+    const STRATEGY_JSON_BODY = 'json body';
+
```

Comment:
Time to get rid "strategies" and polymorphism. Let's do different implementations in inherited classes.